src/HealthChecks.UI.K8s.Operator/Handlers/DeploymentHandler.cs

```diff
@@ -0,0 +1,127 @@
+        }
+
+        public async Task<V1Deployment> GetOrCreateAsync(HealthCheckResource resource)
+        {
+            var deployment = await Get(resource);
+            if (deployment != null) return deployment;
+
+            try
+            {
+                var deploymentResource = Build(resource);
+                var response =
+                    await _client.CreateNamespacedDeploymentWithHttpMessagesAsync(deploymentResource,
+                        resource.Metadata.NamespaceProperty);
+                deployment = response.Body;
+
+                _logger.LogInformation("Deployment {deployment} has been created", deployment.Metadata.Name);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Error creating deployment: {error}", ex.Message);
+            }
+
+            return deployment;
+
+        }
+
+        public async Task Delete(HealthCheckResource resource)
+        {
+            try
+            {
+                await _client.DeleteNamespacedDeploymentAsync($"{resource.Spec.Name}-deploy",
+                    resource.Metadata.NamespaceProperty);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("Error deleting deployment for hc resource: {name} - err: {error}", resource.Spec.Name, ex.Message);
+            }
+        }
+
+        public V1Deployment Build(HealthCheckResource resource)
+        {
+
+            var metadata = new V1ObjectMeta
+            {
+                Labels = new Dictionary<string, string>
+                {
+                    ["resourceId"] = resource.Metadata.Uid,
+                    ["app"] = resource.Spec.Name
+                },
+                Name = $"{resource.Spec.Name}-deploy",
+                NamespaceProperty = resource.Metadata.NamespaceProperty
+            };
+
+            var spec = new V1DeploymentSpec
+            {
+                Selector = new V1LabelSelector
+                {
+                    MatchLabels = new Dictionary<string, string>
+                    {
+                        ["app"] = resource.Spec.Name
+                    }
+                },
+                Replicas = 1,
+                Template = new V1PodTemplateSpec
+                {
+                    Metadata = new V1ObjectMeta
+                    {
+                        Labels = new Dictionary<string, string>
+                        {
+                            ["app"] = resource.Spec.Name
+                        }
+                    },
+                    Spec = new V1PodSpec
+                    {
+                        Containers = new List<V1Container>
+                        {
+                            new V1Container
+                            {
+                                ImagePullPolicy = "Never",
+                                Name = Constants.PodName,
+                                Image = Constants.DockerImage,
```

Comment:
The image should be configurable on the CRD.

Reply:
Why the image should be configurable on the CRD? There is only one image... The docker healthchecks UI image

Reply:
There are many reasons why someone might want a custom image. For example, the Docker images right now will run as root, and fail to run on any cluster with a PodSecurityPolicy that forbids running as root.

Reply:
A custom image created from our base one I guess. Otherwise the operator won't be able to communicate with the UI

Reply:
Exposed a new Image and ImagePullPolicy in the CRD

Reply:
A Docker image can define the user to run as without needing a security context. Running `adduser` and then using the directive `USER newusername` will work if its PodSecurityPolicy has a `MustRunAs` runAsUser rule while a `USER 1000` directive in the Dockerfile will work with both `MustRunAs` and `MustRunAsNonRoot` PodSecurityPolicy runAsUser rules.